### PR TITLE
py-mypy: add 1.3.0

### PR DESCRIPTION
**Pull request by manuelakuhn**
https://github.com/python/mypy/tree/v1.3.0

There was a missing `py-typed-ast` dependency from the [`setup.py`](https://github.com/python/mypy/blob/v1.3.0/setup.py#L224). Since this is a different package than the already contained `py-types-typed-ast`, I have added it.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-mypy/package.py [hunk 1]
```diff
@@ -62,7 +64,7 @@ class PyMypy(PythonPackage):
     depends_on("py-types-typed-ast@1.5.8:1.5", when="@0.981:", type="build")
 
     # setup.py
-    depends_on("python@3.7:", when="@0.981:", type=("build", "run"))
+    depends_on("py-typed-ast@1.4:1", when="@0.920: ^python@:3.7", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mypy/package.py]: This is actually listed above already

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-mypy/package.py (hunk 1 above)
Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-mypy/package.py]: I must have been blind ... Thanks!

**Review by adamjstewart — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-mypy/package.py [hunk 2]
```diff
@@ -61,9 +63,6 @@ class PyMypy(PythonPackage):
     depends_on("py-types-setuptools", when="@0.981:", type="build")
     depends_on("py-types-typed-ast@1.5.8:1.5", when="@0.981:", type="build")
 
-    # setup.py
-    depends_on("python@3.7:", when="@0.981:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mypy/package.py]: Oh yeah, so for some reason we're still supporting Spack being run with Python 3.6, including bootstrapping style dependencies like mypy. So we need to keep this line to get the tests to pass.

Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-mypy/package.py]: OK. Good to know :-)

**Review by adamjstewart — APPROVED** [review]

**Comment by alalazo** [comment]
Need to double check why this one was `auto-merged` even though a test was failing :thinking: Also, I am looking into why this 2 lines change would cause an issue with Python 3.6 on rhel 8

**Comment by alalazo** [comment]
Ok, solved the mystery:
1. Adding this version makes the score for `my-py` worse by 1 (there's a newer version)
2. At that point clingo find a "better" solution by downgrading `flake8`, but upgrading `black`. This configuration is possible because we deleted information for old Python versions (< 3.7) in some packages
3. This new configuration uses flake8 at a version that brings in `flit` and, as a transitive dependency, `docutils`
4. docutils failed because of #38001

**Comment by adamjstewart** [comment]
Can we stop supporting style dependency bootstrapping on EOL versions of Python?

**Comment by alalazo** [comment]
I don't think so, at least until rhel8 is a widely used OS. @tgamblin and @becker33 may have more insight on this.

**Pull request merged**